core/src/genesis.rs

```diff
@@ -323,7 +323,7 @@ pub mod config {
     /// Configuration of the genesis block and the process of its submission.
     pub struct GenesisConfiguration {
         /// The genesis account public key, should be supplied to all peers.
-        /// The type is `Option` just because it might be loaded from environment variables and not from `config.json`.
+        /// The type is `Option` just because it might be loaded from environment variables and not from `config.json`
```

Comment:
I don't understand this comment. Would someone be able to explain it?

Reply:
I think the intention is that `Option` is a workaround for "default keys" request from `Deserialize` (and `Configurable`)

Reply:
aha, if it's not defined in the `config.json` file the deserialization would fail? and it's possible that it's not defined in the `config.json` if it'll be taken from the env variable. What about `cli::Configuration::public_key, should it not also be `Option`?

I think rather than using `Option` there we should have a common deserialization process (one function/method) where fields are tried to be deserialized from a configuration file, if that fails try to deserialize from env variables and fail if it's not possible to deserialize from either. This would require a separate issue

Reply:
I made a change which is in line with what we do elsewhere and that is to use `serde(default)`. This is a security issue as is noted in this #2118

Reply:
We do need a separate issue, because the logic is much more involved than what was just described.